tools/clang/tools/dxcompiler/dxclinker.cpp

```diff
@@ -109,6 +178,8 @@ class DxcLinker : public IDxcLinker, public IDxcContainerEvent {
   std::unique_ptr<DxilLinker> m_pLinker;
   CComPtr<IDxcContainerEventsHandler> m_pDxcContainerEventsHandler;
   std::vector<CComPtr<IDxcBlob>> m_blobs; // Keep blobs live for lazy load.
+  std::map<std::string, const DeserializedDxilCompilerVersion*> m_libNameToCompilerVersionPart;
```

Comment:
Why save ptr here?
If std::set hit resize, the ptr will be invalid.

Reply:
IIRC, it is safe to take pointers to std::set keys, as long as they aren't removed from the set - they will never be reallocated upon resize.  I can't seem to find a reference that explicitly states this right now, however.